procrastinate/sql/migrations/03.00.00_50_post_cancel_notification.sql

```diff
@@ -1,66 +1,103 @@
 DROP TRIGGER IF EXISTS procrastinate_trigger_scheduled_events ON procrastinate_jobs;
+DROP TRIGGER IF EXISTS procrastinate_trigger_status_events_update_v1 ON procrastinate_jobs;
+DROP TRIGGER IF EXISTS procrastinate_jobs_notify_queue_job_inserted_v1 ON procrastinate_jobs;
 
 -- Delete the functions that depends on the old status type
 DROP FUNCTION IF EXISTS procrastinate_fetch_job;
 DROP FUNCTION IF EXISTS procrastinate_finish_job(bigint, procrastinate_job_status, boolean);
 DROP FUNCTION IF EXISTS procrastinate_cancel_job;
-DROP FUNCTION IF EXISTS procrastinate_trigger_status_events_procedure_update;
+DROP FUNCTION IF EXISTS procrastinate_trigger_status_events_procedure_update_v1;
 DROP FUNCTION IF EXISTS procrastinate_finish_job(integer, procrastinate_job_status, timestamp with time zone, boolean);
-
--- Create a new enum type without 'aborting'
-CREATE TYPE procrastinate_job_status_new AS ENUM (
-    'todo',
-    'doing',
-    'succeeded',
-    'failed',
-    'cancelled',
-    'aborted'
-);
-
--- We need to drop the default temporarily as otherwise DatatypeMismatch would occur
-ALTER TABLE procrastinate_jobs ALTER COLUMN status DROP DEFAULT;
+DROP FUNCTION IF EXISTS procrastinate_notify_queue_job_inserted_v1;
 
 -- Alter the table to use the new type
 ALTER TABLE procrastinate_jobs
-ALTER COLUMN status TYPE procrastinate_job_status_new
+ALTER COLUMN status TYPE procrastinate_job_status_v1
 USING (
     CASE status::text
-        WHEN 'aborting' THEN 'doing'::procrastinate_job_status_new
-        ELSE status::text::procrastinate_job_status_new
+        WHEN 'aborting' THEN 'doing'::procrastinate_job_status_v1
+        ELSE status::text::procrastinate_job_status_v1
     END
 );
 
 -- Recreate the default
-ALTER TABLE procrastinate_jobs ALTER COLUMN status SET DEFAULT 'todo'::procrastinate_job_status_new;
-
--- Drop the old type
-DROP TYPE procrastinate_job_status;
-
--- Rename the new type to the original name
-ALTER TYPE procrastinate_job_status_new RENAME TO procrastinate_job_status;
+ALTER TABLE procrastinate_jobs ALTER COLUMN status SET DEFAULT 'todo'::procrastinate_job_status_v1;
 
 -- Recreate the indexes
 CREATE UNIQUE INDEX procrastinate_jobs_queueing_lock_idx ON procrastinate_jobs (queueing_lock) WHERE status = 'todo';
 CREATE UNIQUE INDEX procrastinate_jobs_lock_idx ON procrastinate_jobs (lock) WHERE status = 'doing';
-CREATE INDEX procrastinate_jobs_id_lock_idx ON procrastinate_jobs (id, lock) WHERE status = ANY (ARRAY['todo'::procrastinate_job_status, 'doing'::procrastinate_job_status]);
+CREATE INDEX procrastinate_jobs_id_lock_idx ON procrastinate_jobs (id, lock) WHERE status = ANY (ARRAY['todo'::procrastinate_job_status_v1, 'doing'::procrastinate_job_status_v1]);
+
+-- Drop the old type
+DROP TYPE procrastinate_job_status;
 
--- Recreate and update the functions
-CREATE OR REPLACE FUNCTION procrastinate_fetch_job(
+-- Recreate and update the functions to use the new column
```

Comment:
I don't think this is necessary when just changing the column type. And we already created those functions in `procrastinate/sql/migrations/03.00.00_01_pre_cancel_notification.sql`.